stonybook/pipeline/episode_break_prominence/add_break_prominence.py

```diff
@@ -0,0 +1,189 @@
+import os
+import lxml.etree as ET
+import pickle
+from nltk.corpus import stopwords
+from scipy import signal
+import math
+
+#parse character_coref_annotated.xml and get necessary fields
+def parse_xml(input_xml_path, output_dir):
+    parser = ET.XMLParser(huge_tree=True, remove_blank_text=True)
+    tree = ET.parse(str(input_xml_path), parser=parser)
+    root = tree.getroot()
+
+    lemmas = {}
+    para_breaks = {}
+    chapter_breaks = {}
+    sent_lemma = []
+
+    for mainchild in root:
+        chapterCounter = 0
+        if mainchild.tag == 'body':
+            for var in mainchild:
+                cur = var.attrib
+
+                for paragraphs in list(var):
+                    if paragraphs.tag == 'p':
+                        for sentences in list(paragraphs):
+                            if sentences.tag == 's':
+                                sent_lemma = {}
+                                key = int(sentences.attrib['num'])
+                                sent_lemma[key] = []
+                                for l in list(sentences):
+                                    if l.tag == 't':
+                                        token = int(l.attrib['num'])
+                                        sent_lemma[key].append(l.attrib['lemma'])
+                                    elif l.tag == 'entity':
+                                        for entityLemma in list(l):
+                                            if entityLemma.tag == 't':
+                                                token = int(entityLemma.attrib['num'])
+                                                sent_lemma[key].append(entityLemma.attrib['lemma'])
+
+                                lemmas.update(sent_lemma)
+                        para_breaks[int(paragraphs.attrib['num'])] = key
+
+                if 'desc' in cur.keys():
+                    chapterCounter += 1
+                    chapter_breaks[chapterCounter] = [int(paragraphs.attrib['num']), key]
+
+    with open(os.path.join(output_dir, 'lemmas_pickle.pkl'), 'wb') as f:
```

Comment:
Why do we need to pickle as opposed to passing the object directly to the other function that uses it?

Reply:
Passing lemmas instead of pickle now